Drum Pad Workshop (Resubmission) (#1424)
hackclub/hackclub: workshops/drumpad/README.md

---
name: 'Drum Pad'
description: 'Creating a Drum Pad in HTML, CSS & JS'
author: '@emmanuel39hanks'
---

# Creating a Drum Pad in HTML, CSS & JavaScript

In this workshop, we are going to be creating a drum pad in less than 230 lines of code.

_Preview of the Drum Pad we are going to be creating_

![Drum Pad Preview](https://cloud-ms6ubs6qu.vercel.app/0screencapture-drum-emmanuel39hanks-repl-co-2020-11-07-23_20_37.png)

[Link to demo](https://drum.emmanuel39hanks.repl.co)

## Getting started

Head over to repl.it (https://repl.it/) and we will get started by creating an HTML/CSS & JavaScript repl. Once the repl is set up, navigate to your `index.html` file, and we can start working on the pad layout. 

## Part 1

We are going to start writing HTML; going into our `<body>` tag, which is where we write code that will be shown to the user of the web page. Let's create a header with the `<h1>` tag. Then we will have three rows and four columns of pads and each pad will be a `<div>`. you can think of the `<div>` tag as a box or container.

We will do that with the following code:

> ```html

Reviewer: Maybe consider breaking up the code into smaller chunks? For example, go over the code for the parent div and move the comments into a paragraph outside of the code, rather than including them in the code where they're easy to miss?

<!-- Header tag -->
<h1>DRUM PAD</h1>

<!-- Parent div -->
<div>

<!-- nested child divs -->
  <div>A</div>

  <div>B</div>

  <div>C</div>

  <div>D</div>

  <div>E</div>

  <div>F</div>

  <div>G</div>

  <div>H</div>

  <div>I</div>

  <div>J</div>

  <div>K</div>

  <div>L</div>

</div>

```

And if we run our code, it should look like this:

![Preview of HTML with no CSS](https://cloud-hqtl5tea3.vercel.app/0screencapture-drumpad-emmanuel39hanks-repl-co-2020-11-03-08_36_46.png)

Not so compelling, right? We will now write some CSS, which will style our HTML document and make our drum pad look prettier. Let's do that.

Just before the end of your `<head>` tag, link your CSS file:

```html

<link rel="stylesheet" href="style.css"/>

```

Your CSS file should be linked, and we can start writing our CSS code.

> We will start by writing code for our document to change the colors, adding fonts, height, width, and more. We are also going to be using the font Roboto (https://fonts.google.com/specimen/Roboto?query=roboto), to do that at the top of our CSS file, we will import it using the line `@import 'https://fonts.googleapis.com/css?family=Roboto';`

Reviewer: ```suggestion
We will start by writing code for our document to change the colors, adding fonts, height, width, and more. We are also going to be using the font [Roboto](https://fonts.google.com/specimen/Roboto?query=roboto). To use the font, we will import it using the line `@import 'https://fonts.googleapis.com/css?family=Roboto';`
```

## Part 2

And finally, we are done with our CSS. It now looks good, but when you click on the buttons, we have no sound. We need to write our JavaScript code that will get us sound.

Navigate to the `index.html` file. Then, just before the end of your `<body>` tag, import your `script.js` file that will be used when our document loads:

```html

<script  src="script.js"></script>

```

We will start writing our JavaScript code in the `script.js` file.

```javascript

  // We create a function called play and let it take a parameter which will be a link.

  // A JavaScript function is a block of code designed to perform a particular task. A JavaScript function is executed when "something" invokes it (calls it).

  function  play(link) {

  // Creating an audio variabe and giving it the name audio. To the variable we create a new Audio object which will give us the functionality to sounds, and to the object we pass the link.

  // let is just another keyword that allows you to declare a variable in JavaScript.

  let audio = new Audio(link);

  // We load the sound from the link

  audio.load();

  // And we play it here

  audio.play();

}
```

> What we did above is we created a function called `play()` and it receives a parameter which is `link`, which is the link to the sound hosted on AWS (Amazon Web Services), don't worry, you won't need to host your files by yourself. You can use mine. We then create an audio object and pass the `link` to the object, and then what we do is we just load the audio with the `load()` function and play it with the `play` function.

Reviewer: ```suggestion
What we did above is we created a function called `play()` and it receives a parameter which is `link`. This is the link to the sound hosted on AWS (Amazon Web Services). Don't worry, you won't need to host your files by yourself. You can use mine. We then create an audio object and pass `link` to the object. Now we can just load the audio with the `load()` function and play it with the `play()` function.
```

---
name: 'Drum Pad'
description: 'Creating a Drum Pad in HTML, CSS & JS'
author: '@emmanuel39hanks'
---

Ever wondered how you can play sounds with code? Well, if yes, then you will love this workshop! We will be creating a drum pad in less than 230 lines of code that plays actual sounds, and if you are a beginner, this won't only be fun and creative but will help you get comfortable with coding!

<p align='center'>
<img src="https://media.giphy.com/media/CjmvTCZf2U3p09Cn0h/giphy.gif" alt="ready-to-code" >
</p>

# Overview

_Preview of the Drum Pad we are going to be creating_

![Drum Pad Preview](https://cloud-ms6ubs6qu.vercel.app/0screencapture-drum-emmanuel39hanks-repl-co-2020-11-07-23_20_37.png)

Apart from building the drum pad, you will also be learning about different types of events, functions, styling, and more with Vanilla JavaScript, HTML, and CSS. This whole workshop is customizable according to you, and the best part is, it will take less than 20 minutes to complete! 

[Link to demo](https://drum.emmanuel39hanks.repl.co)

## Getting started

> Let's start by setting up our IDE (Integrated Development Environment) on Rept.it, a free, online code editor.

Reviewer: You mean repl.it, right? Add in a link too.

Author: Yes

> To start your coding right away, navigate to https://repl.it/languages/html, and you will have your whole setup ready when you sign up.

Reviewer: Use markdown link formatting?

Author: Thanks

Inside the `<body>` tag, we will create a big header that displays the text `DRUM PAD`, and we will do that with the `<h1>` tag:

```html

<!-- Header tag -->
<h1>DRUM PAD</h1>
```

Then just under the `<h1>` tag, we will have three rows and four columns of pads, and each pad will be created with the `<div>` tag. You can think of the `<div>` tag as a box or container, and that's why we are using it here because each of our pads will have a boxy look.

```html

<!-- Parent div -->
<div>

<!-- nested child divs -->
  <div>A</div>

  <div>B</div>

  <div>C</div>

  <div>D</div>

  <div>E</div>

  <div>F</div>

  <div>G</div>

  <div>H</div>

  <div>I</div>

  <div>J</div>

  <div>K</div>

  <div>L</div>

</div>

```

To quickly break this down. We have a parent `<div>` tag that nests or holds all of our pads, and then we just pass letters from A to L to easily identify each pad and like I mentioned above, each pad will be given a box look, and that's why we are using `div` tags.

And if we run our code, it should look like this:

![Preview of HTML with no CSS](https://cloud-hqtl5tea3.vercel.app/0screencapture-drumpad-emmanuel39hanks-repl-co-2020-11-03-08_36_46.png)

# CSS Part:

> Not so compelling, right? We will now write some CSS to add stlying, which make our drum pad look prettier. Let's do that!

Reviewer: Worded a bit weirdly, maybe rewrite it as "But this doesn't look that nice, right?"

Author: Sure thing

Try it out by clicking the pads and see the magic!

![Image](https://media.giphy.com/media/11sBLVxNs7v6WA/giphy.gif)
 
## Publishing

Now that you have finished building, you should share your beautiful creation with other people! Remember, it's as easy as giving them your URL! Don't forget to share it with me on Slack @emmanuel39hanks.

## Resources
  
- HTML Docs (https://www.w3schools.com/html/)
- CSS Docs (https://www.w3schools.com/css/)
- JavaScript Docs (https://www.w3schools.com/js/)
- Audio Object (https://www.w3schools.com/JSREF/dom_obj_audio.asp)
- Onclick event (https://www.w3schools.com/jsref/event_onclick.asp)
- JavaScript functions (https://www.w3schools.com/js/js_functions.asp)

## The End

Make sure you have created an account on [repl.it](https://repl.it) to save this wonderful creation!

If you face any difficulties in signing up, [watch this](https://www.youtube.com/watch?v=Mtqp4CUepk0).

Now it is up to you! Do anything with this project, try to implement a way to play the drum pad with your keyboard's corresponding keys. 

Check out these crazy examples!
To finish, here are some amazing, more customized drum pad machines/beatmakers made by other developers:

> - A code playground on Solo Learn by Andrew Siachos (https://code.sololearn.com/WfYyBdZwc6qn/#html)

Reviewer: Aren't you supposed to make specific changes to your code to show how you can change the version you wrote, rather than linking to entirely different versions?

Author: From what I know it is mainly similar projects that are extended or something like that.

  <div  class="box" onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/RS.mp3')">D</div>

  <div  class="box" onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/OH25.mp3')">E</div>

  <div  class="box" onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/MA.mp3')">F</div>

  <div  class="box" onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/CY0010.mp3')">G</div>

  <div  class="box" onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/CH.mp3')">H</div>

  <div  class="box" onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/CB.mp3')">I</div>

  <div  class="box" onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/BD0010.mp3')">J</div>

  <div  class="box"  onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/BD0000.mp3')">K</div>

  <div  class="box"  onclick="play('https://s3-us-west-2.amazonaws.com/s.cdpn.io/242518/CP.mp3')">L</div>

</div>

```

Try it out by clicking the pads and see the magic!
If everything works, your drum pad is complete!

![Image](https://media.giphy.com/media/11sBLVxNs7v6WA/giphy.gif)
 
## Publishing

> 

Reviewer 2: this doesn't need its own section. you should include this + everything in "The End" together inside a section called "Hacking"

Now that you have finished building, you should share your beautiful creation with other people! Remember, it's as easy as giving them your URL! Don't forget to share it with me on Slack @emmanuel39hanks.

> ## Resources

Reviewer 2: instead of being put in this "resources" section at the end, all of these should be linked where appropriate earlier in the workshop as part of an explanation. i don't think the html, css, and javascript docs are directly relevant to the workshop, so they should be removed. the part where you introduce the onclick event in the workshop is a good place to link the onclick event docs. i also don't think you need to include javascript functions unless it's part of an explanation of what functions are within the workshop

  
- HTML Docs (https://www.w3schools.com/html/)
- CSS Docs (https://www.w3schools.com/css/)
- JavaScript Docs (https://www.w3schools.com/js/)
- Audio Object (https://www.w3schools.com/JSREF/dom_obj_audio.asp)
- Onclick event (https://www.w3schools.com/jsref/event_onclick.asp)
- JavaScript functions (https://www.w3schools.com/js/js_functions.asp)

## The End

Make sure you have created an account on [repl.it](https://repl.it) to save this wonderful creation!

> If you face any difficulties in signing up, [watch this](https://www.youtube.com/watch?v=Mtqp4CUepk0).

Reviewer 2: don't include this, it's not totally necessary and people can just ask their club leaders if they're having trouble

Now it is up to you! Do anything with this project, try to implement a way to play the drum pad with your keyboard's corresponding keys. 

Check out these crazy examples!

> To finish, here are some amazing, more customized drum pad machines/beatmakers made by other developers:

Reviewer 2: these two sentences are redundant. one of them should be removed

- A code playground on Solo Learn by Andrew Siachos (https://code.sololearn.com/WfYyBdZwc6qn/#html)

- Music Pad Controller with a Keyboard (https://endertech.com/blog/music-pad-javascript-html-css)

- My favorite one is by Dev Ed, Making Music with JavaScript (https://www.youtube.com/watch?v=8T4SCksjrQ4).

> 

Reviewer 2: i don't think these examples are exactly the right things to include in the workshop. they're cool examples that look kinda like this project, but the demos should be demos made by you or by other hack clubbers that specifically build off the project that you built here. for example, making a meme version of the project, or adding cool effects to each part of the drum pad, or doing audio distortion, or something like that

> We are done with our workshop. Go ahead, customize it, and add more things to it and have fun. I hope you enjoyed this workshop, happy hacking.

Reviewer 2: this is redundant because you said a similar thing earlier
